component/encryptor_v2/encryptor_v2.go

```diff
@@ -0,0 +1,468 @@
+package encryptor_v2
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/binary"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"github.com/Azure/azure-storage-fuse/v2/common/config"
+	"github.com/Azure/azure-storage-fuse/v2/common/log"
+	"github.com/Azure/azure-storage-fuse/v2/internal"
+	"github.com/Azure/azure-storage-fuse/v2/internal/handlemap"
+)
+
+type Encryptor_v2 struct {
```

Comment:
>Encryptor_v2 [](http://example.com/codeflow?start=5&length=12)

You should capture the PR description as a comment in the code also.
encryptor component - Two mount approach with only 1 file for data is used.
4. encryptor_v2 component -  Two mount with 2 files where one will be having cipher text and other metadata used for decryption.